src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -134,10 +135,30 @@ public async Task<HttpResponse> SendRequestAsync(
             logger.Warning("Request retry failed.");
             if (timeoutException != null)
             {
+                //If the correlation id is available, include it in the exception message
+                string msg = "Request to the endpoint timed out.";
+
+                if (headers != null && headers.Count > 0)
+                {
+                    var correlationId = headers[OAuth2Header.CorrelationId];
+                    string correlationIdMsg = headers.ContainsKey(OAuth2Header.CorrelationId) ?
+                                                $" CorrelationId: {correlationId}" :
+                                                string.Empty;
+
+                    var ex = new MsalServiceException(
+                                MsalError.RequestTimeout,
+                                msg + correlationIdMsg,
```

Comment:
suggest adding the " CorrelationId - " as part of the message here. Otherwise it is just an unknown guid.